hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java

```diff
@@ -69,29 +71,34 @@ protected void doUpdateTransaction(Transaction transaction, RecordItem recordIte
 
         var body = recordItem.getTransactionBody().getEthereumTransaction();
         var ethereumDataBytes = DomainUtils.toBytes(body.getEthereumData());
-        var ethereumTransaction = ethereumTransactionParser.decode(ethereumDataBytes);
-
-        // update ethereumTransaction with body values
-        if (body.hasCallData()) {
-            ethereumTransaction.setCallDataId(EntityId.of(body.getCallData()));
+        try {
+            var ethereumTransaction = ethereumTransactionParser.decode(ethereumDataBytes);
+
+            // update ethereumTransaction with body values
+            if (body.hasCallData()) {
+                ethereumTransaction.setCallDataId(EntityId.of(body.getCallData()));
+            }
+
+            // EVM logic uses weibar for gas values, convert transaction body gas values to tinybars
+            convertGasWeiToTinyBars(ethereumTransaction);
+
+            // update ethereumTransaction with record values
+            var transactionRecord = recordItem.getTransactionRecord();
+            ethereumTransaction.setConsensusTimestamp(recordItem.getConsensusTimestamp());
+            ethereumTransaction.setData(ethereumDataBytes);
+            ethereumTransaction.setHash(DomainUtils.toBytes(transactionRecord.getEthereumHash()));
+            ethereumTransaction.setMaxGasAllowance(body.getMaxGasAllowance());
+            ethereumTransaction.setPayerAccountId(recordItem.getPayerAccountId());
+
+            entityListener.onEthereumTransaction(ethereumTransaction);
+            updateAccountNonce(recordItem, ethereumTransaction);
+            recordItem.setEthereumTransaction(ethereumTransaction);
+
+            recordItem.addEntityId(ethereumTransaction.getCallDataId());
+        } catch (InvalidDatasetException e) {
```

Comment:
The parser can throw any number of exceptions that we should handle.

```suggestion
        } catch (RuntimeException e) {
```

Reply:
It looks like `RuntimeException` works given our exception class hierarchy. Some dependencies could throw `NoSuchElementException` or `IndexOutOfBoundsException`, but they too are `RuntimeException`s.